src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -0,0 +1,3393 @@
+            lateral_conv = self.lateral_convs[idx]
+            output_conv = self.output_convs[idx]
+            cur_fpn = lateral_conv(feature)
+            upsampled_feature = nn.functional.interpolate(transformer_encoder_features[-1], size=cur_fpn.shape[-2:], mode='bilinear', align_corners=False)
+            output = cur_fpn + upsampled_feature
+            output = output_conv(output)
+            transformer_encoder_features.append(output)
+
+        for feature in transformer_encoder_features:
+            if num_cur_levels < self.mask2former_num_feature_levels:
+                multi_scale_features.append(feature)
+                num_cur_levels += 1
+
+        # we use the last feature map
+        last_feature_projected = self.mask_projection(transformer_encoder_features[-1])
+
+
+        return Mask2FormerPixelDecoderOutput(
+            last_hidden_state=last_feature_projected,
+            transformer_encoder_features=transformer_encoder_features[0],
+            multi_scale_features=multi_scale_features,
+            hidden_states=tuple(transformer_encoder_features)
+        )
+
+
+class Mask2FormerPixelLevelModule(nn.Module):
+    def __init__(self, config: Mask2FormerConfig):
+        """
+        Pixel Level Module proposed in [Masked-attention Mask Transformer for Universal Image Segmentation](https://arxiv.org/abs/2112.01527). 
+        It runs the input image through a backbone and a pixel decoder, generating an image feature map and pixel embeddings.
+
+        Args:
+            config ([`Mask2FormerConfig`]):
+                The configuration used to instantiate this model.
+        """
+        super().__init__()
+        self.encoder = Mask2FormerSwinTransformerBackbone(config.backbone_config)
+        print("self.encoder.outputs_shapes:",self.encoder.outputs_shapes) #[128, 256, 512, 1024]
+        self.decoder = Mask2FormerPixelDecoder(
+            config=config.pixel_decoder_config,
+            in_features=config.encoder_in_features,
+            feature_size=config.feature_size,
+            mask_feature_size=config.mask_feature_size,
+            feature_channels=self.encoder.outputs_shapes, #[256, 512, 1024], #
+            feature_strides=config.feature_strides, #[8, 16, 32]
+            common_stride=config.common_stride,
+        )
+
+    def forward(self, pixel_values: Tensor, output_hidden_states: bool = False) -> Mask2FormerPixelLevelModuleOutput:
+        features: List[Tensor] = self.encoder(pixel_values)
+        for feature in features:
+            print("pixfeatures:",feature.shape)
+        print("Pixel output_hidden_states:",output_hidden_states)
+        decoder_output: Mask2FormerPixelDecoderOutput = self.decoder(features, output_hidden_states)
+        
+        return Mask2FormerPixelLevelModuleOutput(
+            # the last feature is actually the output from the last layer
+            encoder_last_hidden_state=features[-1],
```

Comment:
Or maybe we can optionally return it?

Reply:
The intention behind returning it here was so that it can be passed to `Mask2FormerModelOutput` inside `Mask2FormerModel` since currently `Mask2FormerModelOutput` has outputs (hidden states) of backbone, pixel decoder and transformer modules.

If we don't return it here then should I remove `encoder_last_hidden_state` from `Mask2FormerModelOutput` as well?

also if we make it optional then can I do it based on value of `output_hidden_states` or should I do it on basis of a new parameter like `output_encoder_hidden_states`?

Reply:
Making it optional sounds good! The idea is to allow users to experiment with different things and make it easier to conduct research.

Reply:
okay makes sense. Will make it optional then.